packages/java/endpoint/src/main/java/com/vaadin/hilla/startup/EndpointUsageDetectorImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.vaadin.hilla.startup;
+
+import com.vaadin.flow.server.frontend.EndpointUsageDetector;
+import com.vaadin.flow.server.frontend.Options;
+import com.vaadin.flow.server.frontend.scanner.ClassFinder;
+import com.vaadin.hilla.BrowserCallable;
+import com.vaadin.hilla.Endpoint;
+
+/**
+ * Implementation of EndpointUsageDetector which determines if the endpoint
+ * generator task needs to be run.
+ */
+public class EndpointUsageDetectorImpl implements EndpointUsageDetector {
+
+    @Override
+    public boolean areEndpointsUsed(Options options) {
+        ClassFinder classFinder = options.getClassFinder();
+        return !classFinder.getAnnotatedClasses(Endpoint.class).isEmpty()
+                || !classFinder.getAnnotatedClasses(BrowserCallable.class)
+                        .isEmpty();
```

Comment:
This check should probably exclude Hilla internal annotated classes, like `SignalsHandler`

Reply:
Excellent point, indeed, without this Flow would always try to generate the endpoints unless one completely excludes Hilla.
Something similar is needed:
```java
        ClassFinder classFinder = options.getClassFinder();
        return Stream.concat(
            classFinder.getAnnotatedClasses(Endpoint.class).stream(),
                classFinder.getAnnotatedClasses(BrowserCallable.class).stream())
            .anyMatch(annotatedClass -> !annotatedClass.getPackage().getName()
                .startsWith("com.vaadin.hilla"));
```